neuralop/layers/gno_block.py

```diff
@@ -0,0 +1,202 @@
+from typing import List
+
+
+import torch
+from torch import nn
+import torch.nn.functional as F
+
+from .channel_mlp import LinearChannelMLP
+from .integral_transform import IntegralTransform
+from .neighbor_search import NeighborSearch
+from .embeddings import SinusoidalEmbedding2D, GridEmbedding2D, GridEmbeddingND
+
+
+class GNOBlock(nn.Module):
+    """GNOBlock implements a Graph Neural Operator layer as described in _[1].
+
+    A GNO layer is a resolution-invariant operator that maps a function defined
+    over one coordinate mesh to another defined over another coordinate mesh using 
+    a pointwise kernel integral that takes contributions from neighbors of distance 1
+    within a graph constructed via neighbor search with a specified radius. 
+
+    Optionally, if provided, the input and output queries can have a positional embedding
+    applied using the argument pos_embedding.
+
+    The kernel integral computed in IntegralTransform 
+    computes one of the following:
+        (a) \int_{A(x)} k(x, y) dy
+        (b) \int_{A(x)} k(x, y) * f(y) dy
+        (c) \int_{A(x)} k(x, y, f(y)) dy
+        (d) \int_{A(x)} k(x, y, f(y)) * f(y) dy
+    
+    Parameters
+    ----------
+    in_channels : int
+        number of channels in input function. Only used if transform_type
+        is (c) "nonlinear" or (d) "nonlinear_kernelonly"
+    out_channels : int
+        number of channels in output function
+    coord_dim : int
+        dimension of domain on which x and y are defined
+    radius : float
+        radius in which to search for neighbors
+    sinusoidal_coord_embed_dim : int, optional
```

Comment:
This seems not to correspond to the signature of the function?

Reply:
Fixed, thanks